### PR TITLE
Initial implementation of  txProcessor

### PR DESCRIPTION
**Pull request by kaiden-araki**
@Shailu-s This is PR, which I have mentioned [before](https://tokamak-network.slack.com/archives/C074PEUC2CR/p1729233403971209?thread_ts=1729129460.765399&cid=C074PEUC2CR).
As you know, it is initial and rough implementation for txProcessor. We can use this as the basic setup for future work related to zki, BatchBuilding, and Synchronization.

### CODE REVIEW

File: go_sequencer/txprocessor/txprocessor.go [hunk 1]
```diff
@@ -153,5 +177,1068 @@ func NewTxProcessor(state *statedb.StateDB, config Config) *TxProcessor {
...
+		acc, err := txProcessor.state.GetAccount(tx.FromIdx)
+		if err != nil {
+			log.Errorw("GetAccount", "fromIdx", tx.FromIdx, "err", err)
+			return nil, nil, false, common.Wrap(err)
+		}
+		tx.Nonce = acc.Nonce
+	}
+
+	switch tx.Type {
+	case common.TxTypeTransfer, common.TxTypeTransferToEthAddr, common.TxTypeTransferToBJJ:
```
Comment (Shailu-s) [inline go_sequencer/txprocessor/txprocessor.go:766]: For L2Tx will we have transfer ? Instead here will there we transactions related to vouch - Create, Delete

Let me know if i missed something ?

File: go_sequencer/synchronizer/synchronizer.go [hunk 2]
```diff
@@ -718,11 +718,11 @@ func (s *Synchronizer) rollupSync(ethBlock *common.Block) (*common.RollupData, e
 				"evtForgeBatch.BatchNum = (%v)",
 				s.stateDB.CurrentBatch(), batchNum))
 		}
-		if s.stateDB.AccountTree.Root().BigInt().Cmp(forgeBatchArgs.NewStRoot) != 0 {
-			return nil, tracerr.Wrap(fmt.Errorf("stateDB.MTRoot (%v) != "+
-				"forgeBatchArgs.NewStRoot (%v)",
-				s.stateDB.AccountTree.Root().BigInt(), forgeBatchArgs.NewStRoot))
-		}
+		// if s.stateDB.AccountTree.Root().BigInt().Cmp(forgeBatchArgs.NewStRoot) != 0 {
```
Comment (Shailu-s) [inline go_sequencer/synchronizer/synchronizer.go:721]: Regarding the commented part here @kaiden-araki Was this intentional. 

If not can you explain the reason. As to me it seems like it's checking the root of account tree to be equal to that received from the `RollupForgeBatchArgs` contract. 

Let me know if i missed someting in between. 

Thanks

File: go_sequencer/txprocessor/txprocessor.go [hunk 3]
```diff
@@ -153,5 +177,1068 @@ func NewTxProcessor(state *statedb.StateDB, config Config) *TxProcessor {
...
+	}
+	// defer txProcessor.resetZKInputs()
+
+	if len(coordIdxs) > int(txProcessor.config.MaxFeeTx) {
+		return nil, common.Wrap(
+			fmt.Errorf("CoordIdxs (%d) length must be smaller than MaxFeeTx (%d)",
+				len(coordIdxs), txProcessor.config.MaxFeeTx))
+	}
+
+	nTx := len(l1usertxs) + len(l1coordinatortxs) + len(l2txs)
```
Comment (Shailu-s) [inline go_sequencer/txprocessor/txprocessor.go:201]: Also, Here @kaiden-araki Can you please explain a bit about the use of L1coordinatortxs in our implementation. Like what will we process, here ? Will this be required for us ?

File: go_sequencer/txprocessor/txprocessor.go (hunk 1 above)
Comment (kaiden-araki) [reply go_sequencer/txprocessor/txprocessor.go:766]: No need of transfer, as you commented we just use vouch related txs.

**Comment by kaiden-araki** [comment]
Hey @Shailu-s, as I said, it is **initial** and **rough** implementation, and there are things to be modified, and you can change as your wish, and we can confirm each other.

**Pull request merged**